src/Symfony/Component/PropertyInfo/Extractor/ReflectionExtractor.php

```diff
@@ -281,7 +300,16 @@ public function getWriteInfo(string $class, string $property, array $context = [
         }
 
         $allowGetterSetter = $context['enable_getter_setter_extraction'] ?? false;
-        $allowMagicCall = $context['enable_magic_call_extraction'] ?? false;
+        $magicMethods = $context['enable_magic_methods_extraction'] ?? $this->magicMethodsFlags;
+        $allowMagicCall = (bool) ($magicMethods & self::ALLOW_MAGIC_CALL);
+        $allowMagicSet = (bool) ($magicMethods & self::ALLOW_MAGIC_SET);
+
+        if (isset($context['enable_magic_call_extraction'])) {
+            trigger_deprecation('symfony/property-info', '5.2', 'Using the "enable_magic_call_extraction" context option in "%s()" is deprecated. Use "enable_magic_methods_extraction" instead.', __METHOD__);
```

Comment:
Other options are keeping this key and have single `enable_magic_get_extraction` / `enable_magic_set_extraction` as well, and either:
- have 3 new constructor arguments for the default
- keep having a single constructor arguments with flags, but keep having dedicated context keys.